core/src/main/java/com/backblaze/b2/json/B2Json.java

```diff
@@ -193,11 +201,11 @@ public String toJson(Object obj) throws B2JsonException {
     }
 
     public String toJson(Object obj, B2JsonOptions options) throws B2JsonException {
-        try (final ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+        try (final B2JsonByteArrayOutputStream out = new B2JsonByteArrayOutputStream()) {
             toJson(obj, options, out);
             return out.toString(B2StringUtil.UTF8);
         } catch (IOException e) {
-            throw new RuntimeException("IO exception writing to string");
+            throw new RuntimeException("IO exception writing to string: " + e.getMessage());
```

Comment:
i _think_ that B2Json is careful to not echo a user's content back in an exception....and that _might_ be why we weren't including the IOException's message in the RuntimeException's message.  can you please take a look to see if we're including the underlying message anywhere else that _might_ contain user-provided content?

Reply:
there were 2 places that do get messages from the caught IO exceptions: toJsonUtf8Bytes, toJsonUtf8BytesWithNewline. Any particular reason for not doing that?

Reply:
thanks for looking.  good question.  i think i don't remember what these messages tend to look like.  we probably thought that they didn't include incoming content and were instead messages like "disk full" or "permission denied".  messages like those seem safe.  the exception you added should be safe.  i looked though and java.io.UnsupportedEncodingException is an IOException and i don't know if it echos back content.

i wish that there wasn't such a thing as "malicious user content".   perhaps it's ok to include this and say it's the responsibility of higher-level software to not leak content (malicious or otherwise).

Reply:
just checked with brianb.  he agrees it's ok to let these through.  this isn't really a problem we can completely solve for software above us.